internal/renderer/renderer.go

```diff
@@ -51,6 +51,21 @@ var (
 // matches a FULL string that contains no non-whitespace characters
 var emptyLineRegex = regexp.MustCompile(`\A[\s]*\z`)
 
+// fairly tolerant to handle weird HTML
+var tagPairRegexString = `<[\n\f ]*%s([\n\f ]+[^\n\f \/>"'=]+[\n\f ]*(=[\n\f ]*([a-zA-Z1-9\-]+|"[^\n\f"]+"|'[^\n\f']+'))?)*[\n\f ]*>.*?<[\n\f ]*/[\n\f ]*%s[\n\f ]*>`
```

Comment:
There's [golang.org/x/net/html](https://pkg.go.dev/golang.org/x/net/html) which could initially simplify the task a bit (but I wonder if its usage will turn into another rabbit hole at some point).

Reply:
Yeah, I could see using this in the future to do better parsing of the file before passing through gomarkdown, to strip out or replace any HTML stuff that gomarkdown just doesn't handle well. At some point I want to start testing this on some real sites that publish their source, and hopefully we'll find out what the most common issues are then!

Reply:
We're probably better served by moving to a better Markdown parser. Hugo moved from Blackfriday to Goldmark since v0.60 anyway.

In retrospect, I have no idea why I picked gomarkdown in the first place, having virtually zero experience with it.